### DISCUSSION TITLE
PBR 2.5

### DISCUSSION THREAD

**Discussion by ghost** [UPBGE/upbge#1736] [Ideas/Features for UPBGE 0.2.5+]
How can we configure a PBR material in UPBGE 2.5? or at least imitate the PBR results, researching I found people who would get very good results but they never tell how they got it.

**Comment by HolandonHdunto** [comment]
Imitate the pbr:
     https://www.youtube.com/watch?v=vPjT20lS53A.

To imitate the PPR is very simple and just use the cubeMap, you take advantage of the luminosity of the cubeMap and multiply it by a base material that will only serve to capture shadows and the light of the lamps and the sun.
To get the roughness and just use Lod Bias to blur the texture.

In short and that:
    https://www.youtube.com/watch?v=4zCuMwG0bIU.

Of course you can also make a cubemap on the fly, but performance is reduced unless you update the cubemap by applying a refresh interval, so instead of the cubemap updating frame by frame, it can be refreshed every 24-24 frames, for example, or update each new scene added.
           Another thing is the scenario, if the scenario is bad, it's no use using the cube map, so it's good to use the cube map together with a realistic cooking scenario.

like this for example:
https://www.youtube.com/watch?v=WLjBASgcQBM
Well, at least in blender 0.3 it has denoise, which means you can bake with reduced Samples.

**Comment by HolandonHdunto** [comment]
And of course if you place multiple materials in the scene it will have a performance drop, I think if you place a material for multiple objects using vertex color, UV map with ramp texture or object color using black color for white, or 0.0 to 1.0 as id, you can select areas of the mesh that will receive the texture, it seems that the exaggerated number of textures in a material makes no difference in the processing.
